tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+			Math.round((y - this.gridOffsetY) / this.gridSize) * this.gridSize +
+			this.gridOffsetY;
+		return { x: snappedX, y: snappedY };
+	}
+
+	_createConvexHull = () => {
+		// Create a convex hull using grahams scan credit: https://en.wikipedia.org/wiki/Graham_scan
+		const qubitStack = [];
+		// Sort the qubits by the leftmost qubit
+		const sortedQubits = this.qubits.toSorted((a, b) => a.globalX - b.globalX);
+		// Using the sorted Qubit find the lowest y value qubit
+
+		let lowestYQubit = sortedQubits[0]; // This is the first qubit in the sorted list
+		// Run a for loop to find the next qubit
+		for (const q in sortedQubits) {
+			if (sortedQubits[q].globalY > lowestYQubit.globalY) {
+				lowestYQubit = sortedQubits[q];
+			}
+		}
+		// Sort points by polar angle with respect to lowestYQubit
+		let sortedPoints = sortedQubits.toSorted((a, b) => {
+			// Calculate the polar angle
+			const polarAngleA = Math.atan2(
+				a.globalY - lowestYQubit.globalY,
+				a.globalX - lowestYQubit.globalX
+			);
+			const polarAngleB = Math.atan2(
+				b.globalY - lowestYQubit.globalY,
+				b.globalX - lowestYQubit.globalX
+			);
+			a.pAngle = polarAngleA;
+			b.pAngle = polarAngleB;
+
+			// Get the distance from the lowestYQubit
+			const distA = Math.sqrt(
+				Math.pow(a.globalX - lowestYQubit.globalX, 2) +
+					Math.pow(a.globalY - lowestYQubit.globalY, 2)
+			);
+			const distB = Math.sqrt(
+				Math.pow(b.globalX - lowestYQubit.globalX, 2) +
+					Math.pow(b.globalY - lowestYQubit.globalY, 2)
+			);
+			a.dist = distA;
+			b.dist = distB;
+			return polarAngleB - polarAngleA;
+		});
+
+		sortedPoints = sortedPoints.filter((qubit) => {
+			// If the qubit is the lowestYQubit, then keep it
+			if (qubit === lowestYQubit) return qubit;
+			// Get the qubits that have the same polar angle
+			const samepQubits = sortedPoints.filter((q) => q.pAngle === qubit.pAngle);
+			// Get the maximum distance qubit
+			const maxDistQubit = samepQubits.sort((a, b) => {
+				return b.dist - a.dist;
+			})[0];
+			// Return the qubit with the maximum distance
+			return qubit === maxDistQubit;
+		});
+		// Create the stack of qubits which contains the convex hull
+		for (const qubit of sortedPoints) {
+			// Remove qubits from the stack if the angle formed by points next-to-top, top, and qubit is not counterclockwise
+			while (
+				qubitStack.length > 1 &&
+				this._ccw(
+					qubitStack.at(-2), // second from top
+					qubitStack.at(-1), // top of stack
+					qubit
+				) <= 0
+			) {
+				qubitStack.pop();
+			}
+			// push the qubit onto the stack
+			qubitStack.push(qubit);
+		}
+		// Map the qubit stack to the global coordinates
+		const qubitPos = qubitStack.map((qubit) => ({
+			x: qubit.globalX,
+			y: qubit.globalY,
+		}));
+
+		this.beginFill(this.color);
+		// Fill the convex hull
+		this.drawPolygon(qubitPos);
+		this.cursor = 'pointer';
+		this.endFill();
+		// Assign the qubit stack to the plaquette
+		this.qubitStack = qubitStack;
```

Comment:
Why don't we perform a sort here, so we don't have to in the getLeftmost, rightmost, etc. functions above every time they are called.

Reply:
can you elaborate on what you would be sorting the qubitStack array? :) thanks

Reply:
If the order of `qubitStack` is irrelevant, outside of those methods, we can keep it sorted so that we don't have to sort again when we want the leftmost/rightmost/etc. qubits. May provide a bit of a performance/simplicity boost.